examples/experimental_component/src/button.rs

```diff
@@ -0,0 +1,97 @@
+#![allow(dead_code)]
+
+use seed::{prelude::*, *};
+use std::borrow::Cow;
+use std::rc::Rc;
+
+pub struct Button<S> {
+    pub label: S,
+}
+
+impl<S: Into<Cow<'static, str>>> Button<S> {
+    pub fn into_component<Ms>(self) -> ButtonComponent<Ms> {
+        ButtonComponent {
+            label: self.label.into(),
+            outlined: false,
+            disabled: false,
+            on_clicks: Vec::new(),
+        }
+    }
+}
+
+#[allow(clippy::module_name_repetitions)]
+pub struct ButtonComponent<Ms: 'static> {
+    label: Cow<'static, str>,
+    outlined: bool,
+    disabled: bool,
+    on_clicks: Vec<Rc<dyn Fn() -> Ms>>,
+}
+
+impl<Ms> ButtonComponent<Ms> {
+    pub const fn outlined(mut self, outlined: bool) -> Self {
+        self.outlined = outlined;
+        self
+    }
+
+    pub const fn disabled(mut self, disabled: bool) -> Self {
+        self.disabled = disabled;
+        self
+    }
+
+    pub fn on_click(mut self, on_click: impl FnOnce() -> Ms + Clone + 'static) -> Self {
+        self.on_clicks.push(Rc::new(move || on_click.clone()()));
+        self
+    }
+}
+
+impl<Ms> Component<Ms> for ButtonComponent<Ms> {
+    fn render(&self) -> Node<Ms> {
```

Comment:
What's the difference between `view` and `render`? I really don't care about the name itself but I like consistent naming ;-)

Reply:
Uhm, yeah. I'm not sure what I was thinking there :) Fixed in https://github.com/seed-rs/seed/pull/599/commits/b57c930724e4a2a457873457d0326f54904af83a